multimodal/src/autogluon/multimodal/data/labelencoder_ner.py

```diff
@@ -0,0 +1,175 @@
+import json
+import logging
+import re
+from typing import Any, Dict, List, Optional, Union
+
+import jsonschema
+import numpy as np
+import pandas as pd
+from nptyping import NDArray
+from omegaconf import DictConfig, OmegaConf
+
+from ..constants import AUTOMM, END_OFFSET, ENTITY_GROUP, NER_ANNOTATION, START_OFFSET
+from .utils import process_ner_annotations
+
+logger = logging.getLogger(AUTOMM)
+
+
+class NerLabelEncoder:
+    """
+    Label Encoder for the named entity recognition task.
+    """
+
+    def __init__(self, entity_map: Optional[dict] = None):
+        self.entity_map = entity_map
+
+    def fit(self, y: pd.Series, x: pd.Series, config: DictConfig):
+        """
+        Extract the annotations, check the unique entity groups, and build entity to index mappings.
+        """
+        model_config = config.model.ner
+        self.ner_special_tags = OmegaConf.to_object(model_config.special_tags)
+        self.prefix = config.model.names[0]
```

Comment:
These three lines seem more suitable in __init__().

Reply:
+1, config should be put in init.

Reply:
it is done in the latest commit